### ISSUE TITLE
“Auto count records” query is firing even when feature disabled

### ISSUE THREAD

**Issue by henrymazza** [JSONAPI-Resources/jsonapi-resources#335]
This code:

``` ruby
            Rails.logger.debug JSONAPI.configuration.inspect
            Customer.
              where([“unaccent(name) ilike unaccent(?)", term])
```

Generates this log:

```
  Customer Load (11.7ms)  SELECT  "customers".* FROM "customers" WHERE (unaccent(name) ilike unaccent('Fabio Mazarotto%'))  ORDER BY "customers"."id" ASC LIMIT 10 OFFSET 0
#<JSONAPI::Configuration:0x007ff09404db70 @json_key_format=:dasherized_key, @key_formatter=DasherizedKeyFormatter, @route_format=:dasherized_route, @route_formatter=DasherizedRouteFormatter, @operations_processor_name=:active_record, @operations_processor=ActiveRecordOperationsProcessor, @allowed_request_params=[:include, :fields, :format, :controller, :action, :sort, :page], @default_paginator=:none, @top_level_links_include_pagination=true, @default_page_size=10, @maximum_page_size=20, @top_level_meta_include_record_count=false, @top_level_meta_record_count_key=:record_count, @use_text_errors=false, @exception_class_whitelist=[], @always_include_to_one_linkage_data=false, @always_include_to_many_linkage_data=false>
   (14.1ms)  SELECT COUNT(*) FROM "customers" WHERE (unaccent(name) ilike unaccent('Fabio Mazarotto%'))
Completed 200 OK in 53ms (Views: 1.0ms | ActiveRecord: 26.0ms)
```

Observe the `@top_level_meta_include_record_count=false` on configuration. The counting isn’t being included in the meta of JSON payload either.

As an aside, the original query had a diferent SELECT to make postgres include the similarity between the attribute and the search term. It blows up when you `.count` the relation. Is there an workaround?

**Comment by lgebhardt** [comment]
Do you by chance have a paginator specified for the resource? The paginators can execute the count query, even if they don't output the result in the metadata.

**Comment by lgebhardt** [comment]
> As an aside, the original query had a diferent SELECT to make postgres include the similarity between the attribute and the search term. It blows up when you .count the relation. Is there an workaround?

I don't have any experience with using `unaccent` in postgresql, but it looks like the query ran from the log. What was different in the original query?

**Comment by henrymazza** [comment]
Ouch! It is the paginator! I disabled it globally to avoid the count but it was re-enabled on this resource. My bad. So the bug is a no-bug. Closing it.

About the breaking stuff. This is my original query from the Rails app I’m porting:

``` SQL
SELECT customers.*, similarity(customers.name, 'Fabio%') AS sim FROM "customers" WHERE (name ilike 'Fabio%' or name % unaccent('Fabio%'))  ORDER BY sim DESC, "customers"."id" ASC
```

If I use ‘autocount’ this is what runs after:

``` SQL
SELECT COUNT(customers.*, similarity(customers.name, 'Fabio%') AS sim) FROM "customers" WHERE (name ilike 'Fabio%')
```

Without autocount it’s working perfectly, it’s even maintaining the sort order. Better if I could paginate and pass the value along as a meta per-record. Perhaps just override the autocount method? 

PS: I’m using `unaccent` to strip the tildes and make “Fabio” match “Fábio”. Isn’t much different than the lowercase operator.
